Tools/machines/perlmutter-nersc/Containerfile

```diff
@@ -0,0 +1,151 @@
+# Base System and Essential Tools Installation
+FROM nvidia/cuda:12.6.0-devel-ubuntu22.04 AS base
+
+# Set up environment variables
+ENV DEBIAN_FRONTEND=noninteractive \
+    SW_DIR=/opt/software \
+    FORCE_UNSAFE_CONFIGURE=1
+
+# Install essential system dependencies including MPI libraries
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    autoconf \
+    build-essential \
+    ca-certificates \
+    coreutils \
+    curl \
+    environment-modules \
+    gfortran \
+    git \
+    openssh-server \
+    python3 \
+    python3-pip \
+    python3-dev \
+    python3-venv \
+    unzip \
+    vim \
+    libmpich-dev \
```

Comment:
Do we need to do anything to get GPU-Aware MPI on Perlmutter to run?
Vanilla without containers, I have to set this to tell the Cray compilers to get it right:
```
# necessary to use CUDA-Aware MPI and run a job
export CRAY_ACCEL_TARGET=nvidia80
```